src/test/kotlin/uk/gov/dwp/dataworks/logging/LogFieldsTest.kt

```diff
@@ -0,0 +1,60 @@
+package uk.gov.dwp.dataworks.logging
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.EnumSource
+import java.lang.IllegalArgumentException
+
+class LogFieldsTest {
+    @ParameterizedTest
+    @EnumSource(LogField::class)
+    fun `Common log fields returns values using LogField object`(logField: LogField) {
+        val actual = LogFields.get(logField)
+        assertThat(actual)
+            .withFailMessage("Expected value of %s to be %s but was %s", logField.systemPropName, logField.default, actual)
+            .isEqualTo(logField.default)
+    }
+
+    @ParameterizedTest
+    @EnumSource(LogField::class)
+    fun `Common log fields returns values using String`(logField: LogField) {
+        val actual = LogFields.get(logField.systemPropName)
+        assertThat(actual)
+            .withFailMessage("Expected value of %s to be %s but was %s", logField.systemPropName, logField.default, actual)
+            .isEqualTo(logField.default)
+    }
+
+    @ParameterizedTest
+    @EnumSource(LogField::class)
+    fun `Common log fields are populated statically`(logField: LogField) {
+        assertThat(LogFields.get(logField.systemPropName)).isEqualTo(logField.default)
+    }
+
+    @ParameterizedTest
+    @EnumSource(LogField::class)
+    fun `LogFields statically populates JSON string`(logField: LogField) {
+        assertThat(LogFields.asJson).contains(logField.systemPropName)
+        assertThat(LogFields.asJson).contains(logField.default)
+        assertThat(LogFields.asJson).contains(""" "${logField.systemPropName}":"${logField.default}" """.trim())
+    }
+
+    @Test
+    fun `Log field can set and retrieve custom values`() {
+        val expectedValue = "customValue"
+        val expectedKey = "customKey"
+        LogFields.put(expectedKey, expectedValue)
+        assertThat(LogFields.get(expectedKey)).isEqualTo(expectedValue)
+    }
+
+    @Test
+    fun `Log field throws exception when custom key is blank`() {
+        assertThrows<IllegalArgumentException> {  LogFields.put("", "value") }
+    }
+
+    @Test
+    fun `Log field throws exception when custom value is blank`() {

```

Comment:
Can we assert the exception text please, so we can see it is meaningful? Otherwise it might as well be "Error"??

Reply:
When we add custom common fields, how can we provide the env var value and system property value and the default? The code only has a Key and a Value, so how does the user of this util make sure the same rules are applied?

Reply:
For point 1 - [same as above](https://github.com/dwp/dataworks-common-logging/pull/1#discussion_r390194553)
For point 2 - I will update `LogFields.put()` to accept a default and resolve in the same way in which `CommonLogFields.resolveFieldValue()` does.

Reply:
A followup commit has been made to consolidate these approaches into a single idiomatic system